File: Java/programmers/dfsbfs/타겟넘버.java
package programmers.dfsbfs;

public class 타겟넘버 {
	 static int[] arr;
	    static int t,answer;
	    static boolean[] visited;
	    public int solution(int[] numbers, int target) {
	        arr = new int[numbers.length];
	        arr = numbers;
	        t = target;
	        dfs(0,0);
	        return answer;
	    }
	    public static void dfs(int cnt ,int temp){
	        if(cnt == arr.length){
	            if(t == temp) answer++;
	            return;
	        }
	            dfs(cnt+1,temp+arr[cnt]);
	            dfs(cnt+1,temp-arr[cnt]);
	    }
}

File: JavaScript/programmers/타겟넘버.js
let answer = 0;
let nums, tar, N;
function solution(numbers, target) {
  nums = numbers;
  tar = target;
  N = numbers.length;
  dfs(0, 0);
  return answer;
}

function dfs(cnt, sum) {
  if (cnt === N) {
    if (sum === tar) answer++;
    return;
  }
  dfs(cnt + 1, sum + nums[cnt]);
  dfs(cnt + 1, sum - nums[cnt]);
}
